src/components/ProgramName.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component, Fragment } from 'react';
+import { Confirm, Input } from 'semantic-ui-react';
+import { connect } from 'react-redux';
+import { hot } from 'react-hot-loader';
+import { withCookies } from 'react-cookie';
+import PropTypes from 'prop-types';
+
+import { changeName as actionChangeName } from '@/actions/code';
+
+const mapStateToProps = ({ code }) => ({ code });
+const mapDispatchToProps = (dispatch, { cookies }) => ({
+  changeName: (id, name) => {
+    const changeNameAction = actionChangeName(id, name, {
+      headers: {
+        Authorization: `JWT ${cookies.get('auth_jwt')}`,
+      },
+    });
+    return dispatch(changeNameAction);
+  },
+});
+
+class Console extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      confirmOpen: false,
+      editingName: null,
+      previousPropName: null, // eslint-disable-line react/no-unused-state
+    };
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    const { code } = props;
+    const { previousPropName } = state;
+
+    if (code.name !== previousPropName) {
+      return {
+        editingName: code.name,
+        previousPropName: code.name,
+      };
+    }
+
+    return null;
+  }
+
+  closeConfirm = () => this.setState({ confirmOpen: false })
+
+  openConfirm = () => this.setState({ confirmOpen: true })
+
+  handleChange = (e) => {
+    this.setState({
+      editingName: e.target.value,
+    });
+  }
+
+  handleSave = () => {
+    const { changeName, code } = this.props;
+    const { editingName } = this.state;
+
+    changeName(code.id, editingName);
+    this.closeConfirm();
+  }
+
+  render() {
+    const { confirmOpen, editingName } = this.state;
+
+    return (
+      <Fragment>
+        <Input
+          type="text"
+          label="Name:"
+          defaultValue={editingName}
+          onChange={this.handleChange}
+          action={{ content: 'Save', onClick: this.openConfirm }}
```

Comment:
It looks like the `Save` button built into Semantic's <Input> object when the `content: 'Save'` is used?

I was curious if the `Save` button could be hidden when `previousPropName !== editingName`, but that could be more trouble than it's worth.